lib/realtime/repo.ex

```diff
@@ -15,4 +15,48 @@ defmodule Realtime.Repo do
       Supervisor.stop(repo)
     end
   end
+
+  @doc """
+  Converts a Postgrex.Result into a given struct
+  """
+  @spec pg_result_to_struct(Postgrex.Result.t(), module()) :: [struct()]
+  def pg_result_to_struct(%Postgrex.Result{rows: rows, columns: columns}, struct) do
+    Enum.map(rows, fn row ->
```

Comment:
Is this how Ecto does it?

Reply:
it's a bit tricky to find but I think so with this spot in the code
https://github.com/elixir-ecto/ecto/blob/8fac7068ed28cbb63c9844bb4cfba09fb48b9a09/lib/ecto/repo/schema.ex#L595

just needs to do this with a ton of code in the middle 😅

Reply:
will check in the elixir slack where does this happen to see if I can reuse it